README.md

```diff
@@ -45,6 +45,18 @@ cargo install rustlings
 
 </details>
 
+> [!CAUTION]
+> Don't try to clone the repository to do the exercises! `rust-analyzer` won't work in that case. Please follow the instructions above instead.
```

Comment:
Updated to your exact wording :white_check_mark: 

I've kept the extra "Why?" background in a collapsible section since it feelslike adds a little bit of something that makes it more understandable and convincing. Happy to remove to get this shipped if you prefer.